server/chatroom.py

```diff
@@ -35,21 +46,104 @@ def saveMessage(cur, sender_id, chatroomID, message_sent):
+        
         cur.execute(
             "INSERT INTO in_chatroom(student_id, uni_id, chatroom_id) VALUES(%(user_id)s, %(uni_id)s, %(chatroom_id)s)",
             {"user_id": user_id, "uni_id": uni_id, "chatroom_id": chatroom_id})
+        
         return True
-    except:
-        return False
+    except BaseException as e:
+        print(f'Exception: {e}')
+        raise HTTPException(
+            status_code=500,
+            detail="Database Error",
+        )
+    
+#Generate invite link for student (for non-course chatroom)
+def GenerateInvite(cur, target_user_id, chatroom_id):
+    try:
+        invite_object = {}
+        invite_object.update({"target_user" : target_user_id})
+        cur.execute("SELECT invite_id FROM chatroom WHERE id = %(chatroom_id)s", {"chatroom_id" : chatroom_id})
+        invite_id = cur.fetchall()[0][0]
+        invite_object.update({"invite_id" : invite_id})
+        return json.dumps(invite_object)
+    except BaseException as e:
+        print(f'Exception: {e}')
+        raise HTTPException(
+            status_code=500,
+            detail="Database Error",
+        )
+
+#Accept an invite and join a (non-course) chatroom
+def AcceptInvite(cur, invite_object, target_user_id):
+    try:
+        invite = json.loads(invite_object)
+
+        
+        #check if target_user_id matches invite["target_user"]
+        if (target_user_id != invite["target_user"]):
+            return False
+        
+        invite_id = invite["invite_id"]
+        
+        cur.execute("SELECT id FROM chatroom WHERE invite_id = %(invite_id)s", {"invite_id" : invite_id})
+        chatroom_id = cur.fetchall()[0][0]
+        
+        #do not allow insertion of user into in_chatroom if uni_id for target user and chatroom do not match
+        cur.execute("SELECT uni_id FROM chatroom WHERE id = %(chatroom_id)s", {"chatroom_id" : chatroom_id})
+        uni_id = cur.fetchall()[0][0]
+        cur.execute("SELECT uni_id FROM student WHERE student_id = %(target_user_id)s", {"target_user_id" : target_user_id})
+    
+        student_uni_id = cur.fetchall()[0][0]
+        
+        if (student_uni_id != uni_id):
+            return False
+        
+        #insert target user into in_chatroom
+        cur.execute("INSERT INTO in_chatroom(student_id, uni_id, chatroom_id) VALUES(%(target_user_id)s, %(uni_id)s, %(chatroom_id)s)",
+                    {"target_user_id" : target_user_id, "uni_id" : uni_id, "chatroom_id" : chatroom_id})
+        
+        return True
+    except BaseException as e:
+        print(f'Exception: {e}')
+        raise HTTPException(
+            status_code=500,
+            detail="Database Error",
+        )
+
+# Get the list of chatrooms that a student is in (for visibility of chatrooms functionality)
+def getChatroomsForStudent(cur, student_id):
+    try:
+        cur.execute("SELECT chatroom_name FROM (in_chatroom JOIN chatroom ON chatroom_id = id) WHERE student_id = %(student_id)s",
```

Comment:
im not sure if just getting the name would be enough since we would prob need the id to show the chatroom if the user clicks into it. @jbekkk wt do u think?